symengine/basic.h

```diff
@@ -86,6 +86,8 @@ enum TypeID {
 #include "basic-methods.inc"
 
 class Basic : public EnableRCPFromThis<Basic> {
+protected:
+    TypeID type_code_;
```

Comment:
[CppCoreGuildelines](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#c133-avoid-protected-data) says avoid `protected` data, I think you shouldn't use `protected` members.